File: frontend/lib/models/patient.dart
class Patient {
  final String? id;
  final String? abhaId;
  final String? name;
  final String? phone;
  final String? gender;
  final DateTime? dateOfBirth;
  final String? address;
  final String? bloodGroup;
  final List<String>? allergies;
  final String? emergencyContact;
  final bool? isMinor;
  final String? guardianName;
  final String? guardianPhone;
  final String? preferredLanguage;

  const Patient({
    this.id,
    this.abhaId,
    this.name,
    this.phone,
    this.gender,
    this.dateOfBirth,
    this.address,
    this.bloodGroup,
    this.allergies,
    this.emergencyContact,
    this.isMinor,
    this.guardianName,
    this.guardianPhone,
    this.preferredLanguage,
  });

  factory Patient.fromJson(Map<String, dynamic> json) => Patient(
        id: json['id'] as String?,
        abhaId: json['abhaId'] as String?,
        name: json['fullName'] as String? ?? json['name'] as String?,
        phone: json['phone'] as String?,
        gender: json['gender'] as String?,
        dateOfBirth: json['dateOfBirth'] != null
            ? DateTime.tryParse(json['dateOfBirth'] as String)
            : null,
        address: json['address'] as String?,
        bloodGroup: json['bloodGroup'] as String?,
        allergies: (json['allergies'] as List?)?.cast<String>(),
        emergencyContact: json['emergencyContact'] as String?,
        isMinor: json['isMinor'] as bool?,
        guardianName: json['guardianName'] as String?,
        guardianPhone: json['guardianPhone'] as String?,
        preferredLanguage: json['preferredLanguage'] as String?,
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'abhaId': abhaId,
        'fullName': name,
        'phone': phone,
        'gender': gender,
        'dateOfBirth': dateOfBirth?.toIso8601String(),
        'address': address,
        'bloodGroup': bloodGroup,
        'allergies': allergies,
        'emergencyContact': emergencyContact,
        'isMinor': isMinor,
        'guardianName': guardianName,
        'guardianPhone': guardianPhone,
        'preferredLanguage': preferredLanguage,
      };

  Patient copyWith({
    String? id,
    String? abhaId,
    String? name,
    String? phone,
    String? gender,
    DateTime? dateOfBirth,
    String? address,
    String? bloodGroup,
    List<String>? allergies,
    String? emergencyContact,
    bool? isMinor,
    String? guardianName,
    String? guardianPhone,
    String? preferredLanguage,
  }) =>
      Patient(
        id: id ?? this.id,
        abhaId: abhaId ?? this.abhaId,
        name: name ?? this.name,
        phone: phone ?? this.phone,
        gender: gender ?? this.gender,
        dateOfBirth: dateOfBirth ?? this.dateOfBirth,
        address: address ?? this.address,
        bloodGroup: bloodGroup ?? this.bloodGroup,
        allergies: allergies ?? this.allergies,
        emergencyContact: emergencyContact ?? this.emergencyContact,
        isMinor: isMinor ?? this.isMinor,
        guardianName: guardianName ?? this.guardianName,
        guardianPhone: guardianPhone ?? this.guardianPhone,
        preferredLanguage: preferredLanguage ?? this.preferredLanguage,
      );
}

File: backend/src/main/java/com/medikiosk/model/entity/Patient.java
package com.medikiosk.model.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "patients")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Patient {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "abha_id", unique = true)
    private String abhaId;

    @Column(name = "full_name", nullable = false)
    private String fullName;

    @Column(name = "date_of_birth")
    private LocalDate dateOfBirth;

    private String gender;
    private String phone;
    private String address;

    @Builder.Default
    @Column(name = "is_minor")
    private Boolean isMinor = false;

    @Column(name = "guardian_phone")
    private String guardianPhone;

    @Builder.Default
    @Column(name = "preferred_language")
    private String preferredLanguage = "en";

    @Column(name = "aadhaar_last_four")
    private String aadhaarLastFour;

    @Column(name = "face_embedding", columnDefinition = "TEXT")
    private String faceEmbedding;

    @Builder.Default
    @Column(name = "fingerprint_enrolled")
    private Boolean fingerprintEnrolled = false;

    @Column(name = "enrolled_device_id")
    private String enrolledDeviceId;

    @Column(name = "biometric_enrolled_at")
    private LocalDateTime biometricEnrolledAt;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    
    public UUID getId() { return id; }
    public String getAbhaId() { return abhaId; }
    public String getFullName() { return fullName; }
    public LocalDate getDateOfBirth() { return dateOfBirth; }
    public String getGender() { return gender; }
    public String getPhone() { return phone; }
    public String getAddress() { return address; }
    public Boolean getIsMinor() { return isMinor; }
    public String getGuardianPhone() { return guardianPhone; }
    public String getPreferredLanguage() { return preferredLanguage; }
    public String getAadhaarLastFour() { return aadhaarLastFour; }
    public String getFaceEmbedding() { return faceEmbedding; }
    public Boolean getFingerprintEnrolled() { return fingerprintEnrolled; }
    public String getEnrolledDeviceId() { return enrolledDeviceId; }
    public LocalDateTime getBiometricEnrolledAt() { return biometricEnrolledAt; }
    public LocalDateTime getCreatedAt() { return createdAt; }
    public LocalDateTime getUpdatedAt() { return updatedAt; }

    public void setId(UUID id) { this.id = id; }
    public void setAbhaId(String abhaId) { this.abhaId = abhaId; }
    public void setFullName(String fullName) { this.fullName = fullName; }
    public void setDateOfBirth(LocalDate dateOfBirth) { this.dateOfBirth = dateOfBirth; }
    public void setGender(String gender) { this.gender = gender; }
    public void setPhone(String phone) { this.phone = phone; }
    public void setAddress(String address) { this.address = address; }
    public void setIsMinor(Boolean isMinor) { this.isMinor = isMinor; }
    public void setGuardianPhone(String guardianPhone) { this.guardianPhone = guardianPhone; }
    public void setPreferredLanguage(String preferredLanguage) { this.preferredLanguage = preferredLanguage; }
    public void setAadhaarLastFour(String aadhaarLastFour) { this.aadhaarLastFour = aadhaarLastFour; }
    public void setFaceEmbedding(String faceEmbedding) { this.faceEmbedding = faceEmbedding; }
    public void setFingerprintEnrolled(Boolean fingerprintEnrolled) { this.fingerprintEnrolled = fingerprintEnrolled; }
    public void setEnrolledDeviceId(String enrolledDeviceId) { this.enrolledDeviceId = enrolledDeviceId; }
    public void setBiometricEnrolledAt(LocalDateTime biometricEnrolledAt) { this.biometricEnrolledAt = biometricEnrolledAt; }
}